pkg/apis/application/v1alpha1/types.go

```diff
@@ -56,7 +56,7 @@ type Application struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
 	Spec              ApplicationSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
-	Status            ApplicationStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+	Status            ApplicationStatus `json:"status,omitempty,omitzero" protobuf:"bytes,3,opt,name=status"`
```

Comment:
This syntax isn't still valid since the Go version that Argo CD uses is Go.123.x

Reply:
You are correct @sivchari, this PR would also need a bump to 1.24 (or create a PR upgrading Golang to 1.24)

Reply:
@blakepettersson 
Originally, I'm trying to upgrade to Go1.23.6 to avoid vulnerability in
https://github.com/argoproj/argo-cd/pull/21929

But it seems like a good to upgrade to latest version.
So I do it in above PR.

Reply:
When https://github.com/argoproj/argo-cd/pull/22242 is merged, this PR will pass.